airflow/www/templates/airflow/ti_log.html

```diff
@@ -29,6 +29,28 @@ <h4>{{ title }}</h4>
     </a>
   </li>
   {% endfor %}
+  {% if tailing_required %}
+    <ul class="nav nav-pills pull-right">
+      <li>
+        <div class="dropdown">
+          <button class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" id="tailLinesList">
+            <span id="selected">Tail Logs</span><span class="caret"></span>
+          </button>
+          <ul class="dropdown-menu" aria-labelledby="tailLinesList" id="linesList">
+            {% for tail_line in tail_lines_list %}
```

Comment:
I'm not a fan of this being shown for log storage methods (such as S3/GCP) that don't support this.

Reply:
Changes have been made to stop showing dropdown for unsupported platforms. Please review.

Reply:
They could be made to support it (in the future?) Both S3 and GCS support `Range` headers to fetch specific parts of a file/object.